pkg/kubelet/prober/worker.go

```diff
@@ -188,6 +188,13 @@ func (w *worker) doProbe() (keepGoing bool) {
 		return false
 	}
 
+	// We do not need to check status during deletion
+	if w.pod.ObjectMeta.DeletionTimestamp != nil {
```

Comment:
is it guaranteed that will the traffic already be stopped to this pod when the `DeletionTimestamp ` was set? if not, would it be actually helpful to actually run the readyness probe and take the pod from rotation for the duration of pod termination?

Reply:
This is exactly what will happen once `return false` is executed 3 lines below.

Reply:
My question is what mechanism will remove traffic (not the probes calls, regular user traffic) to this pod? Is it done just by `kubectl delete` somehow before setting this `DeletionTimestamp`?

Looking below I see that in case of terminated container we also do this:

```
		if !w.containerID.IsEmpty() {
			w.resultsManager.Set(w.containerID, results.Failure, w.pod)
		}
```

Same comment below for the unit tests. The question is whether ready prob status needs to be changed while pod is being deleted.

Reply:
Hmm I think the readiness probe failing during deletion does makes sense since it is possible for a service to continue to send traffic to the pod even after deletion. https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/endpoint/endpoints_controller.go#L425-L428 which I was unaware of. 

So maybe this does not fix the bug that I am experiencing, but maybe there's another solution? 

What my system is experiencing is when a pod takes a bit of time to delete it is in the state running, but while clean its self up the readiness probe starts failing (this pod is not behind a service) once the readiness probe fails however it changes the `Phase` of the pod from `Running` to `Pending` even though it has a `deletionTimestamp` set.

My first thought was the readiness probe should not fail since it's being deleted, but that seems to not be the case, however changing the status from `Running` -> `Pending` also does not feel correct.

Reply:
Moving from Running -> pending is definitely a bug.

Reply:
I think `Running` and `Pending` are set by the container runtime... nothing to do with probes.

Reply:
Running -> Pending should be fixed by https://github.com/kubernetes/kubernetes/pull/95364 and https://github.com/kubernetes/kubernetes/pull/95561